File: bindings/java-jni/src/main/java/org/maplibre/nativejni/resource/ResourceKind.java
package org.maplibre.nativejni.resource;

/** Resource kind reported to runtime resource callbacks. */
public enum ResourceKind {
  UNKNOWN(0),
  STYLE(1),
  SOURCE(2),
  TILE(3),
  GLYPHS(4),
  SPRITE_IMAGE(5),
  SPRITE_JSON(6),
  IMAGE(7);

  private final int nativeValue;

  ResourceKind(int nativeValue) {
    this.nativeValue = nativeValue;
  }

  public int nativeValue() {
    return nativeValue;
  }

  public static ResourceKind fromNative(int nativeValue) {
    for (var kind : values()) {
      if (kind.nativeValue == nativeValue) {
        return kind;
      }
    }
    return UNKNOWN;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/resource/ResourceKind.kt
package org.maplibre.nativeffi.resource

/** Resource kind reported to runtime resource callbacks. */
public enum class ResourceKind(public val nativeValue: Int) {
  UNKNOWN(0),
  STYLE(1),
  SOURCE(2),
  TILE(3),
  GLYPHS(4),
  SPRITE_IMAGE(5),
  SPRITE_JSON(6),
  IMAGE(7);

  public companion object {
    internal fun fromNative(nativeValue: UInt): ResourceKind = fromNative(nativeValue.toInt())

    public fun fromNative(nativeValue: Int): ResourceKind =
      entries.firstOrNull { it.nativeValue == nativeValue } ?: UNKNOWN
  }
}
